src/components/Capture/FaceCaptureFlow.js

```diff
@@ -0,0 +1,40 @@
+import { h, Component } from 'preact'
+
+import VideoIntro from '../Video/Intro'
+import FaceCapture from './Face'
+import withCameraDetection from './withCameraDetection'
+import withCrossDeviceWhenNoCamera from './withCrossDeviceWhenNoCamera'
+import { compose } from '~utils/func'
+
+class FaceCaptureFlow extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      displayView: "intro",
+    }
+  }
+
+  displayFaceCaptureStep = () => {
+    this.setState({
+      displayView: "faceCapture"
+    })
+  }
+
+  render() {
+    const { hasCamera, requestedVariant, trackScreen } = this.props
+    if (hasCamera === null) return false
+
+    if (hasCamera && requestedVariant === 'video' && this.state.displayView === "intro") {
+      return <VideoIntro trackScreen={ trackScreen } continueFlow={ this.displayFaceCaptureStep } />
```

Comment:
I think that this logic should belong to the video flow components. If a video can be taken then the video flow should display the intro. The `Face` component should only return either Selfie, Video or uploader.

Reply:
Good point. 
My thinking was to have this as a controller sitting between Capture and Face that decides whether the Intro or Face component should be displayed based on camera availability. But I guess it can be determined at Video (or Camera when that has an Intro too) level.